src/Orleans.Reminders/GrainReminderExtensions.cs

```diff
@@ -0,0 +1,173 @@
+    /// <returns>Completion promise for this operation.</returns>
+    public static Task UnregisterReminder(this IGrainBase grain, IGrainReminder reminder)
+    {
+#if NET6_0_OR_GREATER
+    ArgumentNullException.ThrowIfNull(grain, nameof(grain));
+#else
+        if (reminder is null)
+        {
+            throw new ArgumentNullException(nameof(reminder));
+        }
+#endif
+
+        EnsureRuntime();
+
+        return GetReminderService(grain)
+            .UnregisterReminder(reminder);
+    }
+
+    /// <summary>
+    /// Returns a previously registered reminder.
+    /// </summary>
+    /// <param name="grain">The grain instance.</param>
+    /// <param name="reminderName">Reminder to return</param>
+    /// <returns>Promise for Reminder handle.</returns>
+    public static Task<IGrainReminder> GetReminder(this Grain grain, string reminderName) => GetReminder((IGrainBase)grain, reminderName);
+
+    /// <summary>
+    /// Returns a previously registered reminder.
+    /// </summary>
+    /// <param name="grain">A grain.</param>
+    /// <param name="reminderName">Reminder to return</param>
+    /// <returns>Promise for Reminder handle.</returns>
+    public static Task<IGrainReminder> GetReminder(this IGrainBase grain, string reminderName)
+    {
+#if NET6_0_OR_GREATER
+    ArgumentNullException.ThrowIfNull(grain, nameof(grain));
+#else
+        if (grain is null)
+        {
+            throw new ArgumentNullException(nameof(grain));
+        }
+#endif
+
+        if (string.IsNullOrWhiteSpace(reminderName))
+        {
+            throw new ArgumentNullException(nameof(reminderName));
+        }
+
+        EnsureRuntime();
+
+        return GetReminderService(grain)
+            .GetReminder(grain.GrainContext.GrainReference, reminderName);
+    }
+
+    /// <summary>
+    /// Returns a list of all reminders registered by the grain.
+    /// </summary>
+    /// <returns>Promise for list of Reminders registered for this grain.</returns>
+    public static Task<List<IGrainReminder>> GetReminders(this Grain grain) => GetReminders((IGrainBase)grain);
+
+    /// <summary>
+    /// Returns a list of all reminders registered by the grain.
+    /// </summary>
+    /// <returns>Promise for list of Reminders registered for this grain.</returns>
+    public static Task<List<IGrainReminder>> GetReminders(this IGrainBase grain)
+    {
+#if NET6_0_OR_GREATER
+    ArgumentNullException.ThrowIfNull(grain, nameof(grain));
+#else
+        if (grain is null)
+        {
+            throw new ArgumentNullException(nameof(grain));
+        }
+#endif
+        EnsureRuntime();
+
+        return GetReminderService(grain).GetReminders(grain.GrainContext.GrainReference);
+    }
+
+    /// <summary>
+    /// Gets the <see cref="IReminderService"/>.
+    /// </summary>
+    private static IReminderService GetReminderService(IGrainBase grain) => grain.GrainContext.ActivationServices.GetRequiredService<IReminderService>();
```

Comment:
Ah, here's the problem: this is directly accessing the `IReminderService` instead of going through `IReminderRegistry` - so it's pointing to the wrong reminder service instance, always pointing to the local instance instead of the instance which is responsible for the grain.

Reply:
I'll push an update

Reply:
Thanks for working through this. Appreciate it!